src/extension/rn-extension.ts

```diff
@@ -5,64 +5,93 @@ import {FileSystem} from "../common/node/fileSystem";
+}
 
-    // Asynchronously enable telemetry
-    Telemetry.init("react-native", require("../../package.json").version, true)
+function setupReactDir(context: vscode.ExtensionContext): Q.Promise<void> {
+    const reactDirManager = new ReactDirManager();
+    return reactDirManager.create()
         .then(() => {
-            const reactNativeProjectHelper = new ReactNativeProjectHelper(workspaceRootPath);
-            return reactNativeProjectHelper.isReactNativeProject()
-                .then(isRNProject => {
-                    if (isRNProject) {
-                        reactNativeProjectHelper.validateReactNativeVersion().fail(reason => {
-                            TelemetryHelper.sendSimpleEvent("launchDebuggerError", { rnVersion: reason });
-                            const shortMessage = `React Native Tools only supports React Native versions 0.19.0 and later`;
-                            const longMessage = `${shortMessage}: ${reason}`;
-                            vscode.window.showWarningMessage(shortMessage);
-                            let output = vscode.window.createOutputChannel("React-Native");
-                            output.appendLine(longMessage);
-                            output.show();
-                        }).done();
-                        setupReactNativeDebugger();
-                        IntellisenseHelper.setupReactNativeIntellisense();
-                        context.subscriptions.push(new ReactDirManager());
-                    }
-                }).then(() => {
-                    // Register React Native commands
-                    context.subscriptions.push(vscode.commands.registerCommand("reactNative.runAndroid",
-                        () => commandPaletteHandler.runAndroid()));
-                    context.subscriptions.push(vscode.commands.registerCommand("reactNative.runIos",
-                        () => commandPaletteHandler.runIos()));
-                    context.subscriptions.push(vscode.commands.registerCommand("reactNative.startPackager",
-                        () => commandPaletteHandler.startPackager()));
-                    context.subscriptions.push(vscode.commands.registerCommand("reactNative.stopPackager",
-                        () => commandPaletteHandler.stopPackager()));
+            context.subscriptions.push(reactDirManager);
+        });
+}
 
-                    const nodeDebugPath = vscode.extensions.getExtension("andreweinand.node-debug").extensionPath;
-                    const fsUtil = new FileSystem();
-                    fsUtil.writeFile(path.resolve(__dirname, "../", "debugger", "nodeDebugLocation.json"), JSON.stringify({ nodeDebugPath })).done();
-                });
-        }).done();
+function warnWhenReactNativeVersionIsNotSupported(): void {
+    return reactNativeProjectHelper.validateReactNativeVersion().done(() => { }, reason => {
+        TelemetryHelper.sendSimpleEvent("launchDebuggerError", { rnVersion: reason });
```

Comment:
Should this be a more generic telemetry event, rather than "launchDebuggerError"?